Sources/NIOPosix/PendingWritesManager.swift

```diff
@@ -97,12 +97,23 @@ internal enum OneWriteOperationResult {
 /// The result of trying to write all the outstanding flushed data. That naturally includes all `ByteBuffer`s and
 /// `FileRegions` and the individual writes have potentially been retried (see `WriteSpinOption`).
 internal struct OverallWriteResult {
-    enum WriteOutcome {
+    enum WriteOutcome: Equatable {
         /// Wrote all the data that was flushed. When receiving this result, we can unsubscribe from 'writable' notification.
-        case writtenCompletely
+        case writtenCompletely(CloseState)
 
         /// Could not write everything. Before attempting further writes the eventing system should send a 'writable' notification.
         case couldNotWriteEverything
+
+        static func == (lhs: Self, rhs: Self) -> Bool {
+            switch (lhs, rhs) {
+            case (.writtenCompletely, .writtenCompletely):
+                return true
+            case (.couldNotWriteEverything, .couldNotWriteEverything):
+                return true
+            default:
+                return false
+            }
+        }
```

Comment:
I get a bit nervous about ignoring state when comparing. Do we _actually_ need the equality check? Also, it might make more sense to do this on the `CloseState` and compare the identity of its underlying futures where applicable.

Reply:
It turns out we don't need it anymore so I've just deleted it.

Reply:
🤦‍♂️ it's needed in tests. Looking again.

Reply:
I think this has exposed that I was exposing the promise in a way which wasn't needed anyway because we just passed it right into `pendingWrites.close` which already held it. I've changed the method to return the `CloseResult` instead.